bundles/org.openhab.binding.hue/src/main/java/org/openhab/binding/hue/internal/handler/HueLightHandler.java

```diff
@@ -493,36 +509,35 @@ public void onLightStateChanged(@Nullable HueBridge bridge, FullLight fullLight)
             updateState(CHANNEL_ALERT, stringType);
             scheduleAlertStateRestore(stringType);
         }
+
+        return true;
     }
 
     @Override
     public void channelLinked(ChannelUID channelUID) {
-        HueClient handler = getHueClient();
-        if (handler != null) {
-            FullLight light = handler.getLightById(lightId);
-            if (light != null) {
-                onLightStateChanged(null, light);
-            }
+        final FullLight light = lastFullLight;
+        if (light != null) {
+            onLightStateChanged(null, light);
         }
     }
 
     @Override
     public void onLightRemoved(@Nullable HueBridge bridge, FullLight light) {
-        if (light.getId().equals(lightId)) {
+        if (light.getId().equals(lightId)) { // TODO Maybe this check is obsolete
             updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.NONE, "@text/offline.light-removed");
         }
     }
 
     @Override
     public void onLightGone(@Nullable HueBridge bridge, FullLight light) {
-        if (light.getId().equals(lightId)) {
+        if (light.getId().equals(lightId)) { // TODO Maybe this check is obsolete
             updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.GONE, "@text/offline.light-not-reachable");
         }
     }
 
     @Override
     public void onLightAdded(@Nullable HueBridge bridge, FullLight light) {
-        if (light.getId().equals(lightId)) {
+        if (light.getId().equals(lightId)) { // TODO Maybe this check is obsolete
```

Comment:
With your changes, this check becomes useless (it was useful before).

Reply:
Removed